Singularity/Singularity/Screen/Button.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+using Singularity.Input;
+using Singularity.Property;
+
+
+namespace Singularity.Screen
+{
+    /// <summary>
+    /// Button class allows buttons to be created using either
+    /// text or Texture2D. The buttons can send out Clicked, Released,
+    /// and Hovering events that other classes can subscribe to.
+    /// </summary>
+    class Button : IUpdate, IDraw //TODO this needs to implement IWindowItem which will have update and draw
```

Comment:
Why not already do this TODO? does IWindowItem not already exist?

Reply:
added the interface